tests/bin/get_att.c

```diff
@@ -9,7 +15,12 @@ int main(void) {
 
     ssize_t size = get_att(nonce, sizeof(nonce), buf, sizeof(buf), &technology);
 
-    // TODO: Currently this should return ENOSYS. However, this will change
-    // as get_att() is implemented for each technology.
-    return !(size == -1 && errno == ENOSYS);
+    if (size >= 0)
+	return !(NO_KEEP <= technology <= SEV_TECH);
```

Comment:
```suggestion
    if (size >= 0) {
	switch (technology) {
	case NO_KEEP:
	case SEV:
	case SGX: 
	    /* fallthrough is intentional */
	    return 0;
	default: return 1;
	}
    }
```

I think something like the above is easier to follow because it requires less mental gymnastics for operator precedence.

Reply:
Also I think the boolean logic the way it is right now will always unconditionally return 0

```
return !(NO_KEEP <= technology <= SEV_TECH)
return !((0 <= 0..INTMAX) <= 0) 
return !(1 <= 1)
return !(1)
return 0
```

`technology` being unsigned means it will never be less than `NO_KEEP`

Reply:
Ooh, good catch. I had forgotten `technology` is unsigned.